src/test/java/org/springframework/samples/petclinic/vet/VetTests.java

```diff
@@ -40,4 +45,37 @@ public void testSerialization() {
         assertThat(other.getId()).isEqualTo(vet.getId());
     }
 
+    @Test
+    public void testGetSpecialties(){
+        Vet vet = new Vet();
+        vet.setFirstName("Zaphod");
+        vet.setLastName("Beeblebrox");
+        vet.setId(123);
+        Specialty radiology = new Specialty();
+        Specialty urinalysis = new Specialty();
+        Specialty kinesiology = new Specialty();
+        vet.addSpecialty(radiology);
+        vet.addSpecialty(urinalysis);
+        //vet.addSpecialty(kinesiology);
+
+        List<Specialty> testList = vet.getSpecialties();
+        assertThat(testList).isEqualTo(vet.getSpecialties());
+        testList.getClass().getSimpleName().equals("UnmodifiableCollection");
+    }
+
+    @Test
+    public void testAddGetNrSpecialty(){
+        Vet vet = new Vet();
+        vet.setFirstName("Zaphod");
+        vet.setLastName("Beeblebrox");
+        vet.setId(123);
+
+        int preAddedNumber = vet.getNrOfSpecialties();
+        Specialty radiology = new Specialty();
+        //radiology.setName("radiology");
+        vet.addSpecialty(radiology);
+        assertThat(vet.getNrOfSpecialties()).isEqualTo(++preAddedNumber);
+        assertThat((vet.getSpecialtiesInternal()).contains(radiology));
```

Comment:
`assertThat(vet.getNrOfSpecialties()).isEqualTo(++preAddedNumber)` and `assertThat((vet.getSpecialtiesInternal()).contains(radiology))` should be in their own separate test case


In addition, since `vet.getSpecialtiesInternal` returns a `Set<Specialty>`, it would be useful to check if `radiology` is within that set, perhaps by using getName or something. Basically, wha jad said.